### PR TITLE
Feature/us6362 host edits their gathering

### PR DESCRIPTION
**Pull request by JoeKer1**
Lots of refactoring and host and person edit forms.

### CODE REVIEW

File: src/app/components/add-me-to-map/add-me-to-map.component.html [hunk 1]
```diff
@@ -12,7 +12,17 @@
 
     <div class="fauxdal-content">
       <crds-content-block id="finderAddMeToMapText"></crds-content-block>
-      <address-form class="push-top" [userData]="userData" (save)="onSubmit($event)"></address-form>
+      <form novalidate [formGroup]="addMeToMapForm" (ngSubmit)="onSubmit(addMeToMapForm)">
+        <address-form class="push-top"[parentForm]="addMeToMapForm" groupName="addressForm" [isFormSubmitted]="submissionError" [address]="userData.address"></address-form>
```
Comment (ilpadre) [inline src/app/components/add-me-to-map/add-me-to-map.component.html]: Should have a space between "push-top" and [parentForm]

File: src/app/components/pin-details/gathering/gathering.component.spec.ts [hunk 2]
```diff
@@ -66,6 +67,7 @@ describe('GatheringComponent', () => {
         mockToast = jasmine.createSpyObj<ToastsManager>('toast', ['warning', 'error']);
         mockContentService = jasmine.createSpyObj<ContentService>('contentService', ['getContent']);
         mockAngulartics2 = new MockAngulartic();
+        mockRouter = jasmine.createSpyObj<Router>('router', ['navigate']);
```
Comment (ilpadre) [inline src/app/components/pin-details/gathering/gathering.component.spec.ts]: Is this necessary given the route is mocked below?

Comment (JoeKer1) [inline src/app/components/pin-details/gathering/gathering.component.spec.ts]: nope it's not i was trying to get it to work and then did that instead.

File: src/app/components/pin-details/person/person.component.spec.ts [hunk 3]
```diff
@@ -22,13 +23,14 @@ describe('PersonComponent', () => {
...
 
-    let mockContentService, mockToast, mockStateService, mockAddressService;
+    let mockContentService, mockToast, mockStateService, mockAddressService, mockRouter;
 
     beforeEach(() => {
         mockContentService = jasmine.createSpyObj<ContentService>('content', ['getContent']);
         mockAddressService = jasmine.createSpyObj<AddressService>('addressService', ['getFullAddress']);
         mockToast = jasmine.createSpyObj<ToastsManager>('toast', ['warning', 'error']);
         mockStateService = jasmine.createSpyObj<StateService>('state', ['setLoading', 'setPageHeader']);
+        mockRouter = jasmine.createSpyObj<Router>('router', ['navigatge']);
```
Comment (ilpadre) [inline src/app/components/pin-details/person/person.component.spec.ts]: What's navigatge?

**Pull request merged**